### PR TITLE
JEST-106 update: db_migrate no longer relies on host mysql binary

### PR DESCRIPTION
**Pull request by chavezellis**

### CODE REVIEW

File: scripts/db_migrate.sh [hunk 1]
```diff
@@ -8,14 +8,15 @@ DB_PORT=''
...
 DB_USER='root'
 DB_PASS='root'
-DB_EXPORT_FLAGS='--routines --column-statistics=0 --quick --hex-blob --single-transaction'
-DB_IMPORT_FLAGS=''
+DB_EXPORT_FLAGS='--routines --quick --hex-blob --single-transaction'
 
 # Determine DB
 echo "Welcome to the SDIT DB Migrator (mysql/postgres) $NEWLINE"
 read -rp "Select (mysql|postgres):" DB_MODE
+read -rp "Select (FROM) docker container name: " CONTAINER_NAME_FROM
```
Comment (iBotPeaches) [inline scripts/db_migrate.sh:16]: I don't think we need these asks. It just makes this command harder to use. If you take a look at `docker ps`

```
5db86a7f4510   mariadb:10.3                    "docker-entrypoint.s…"   2 weeks ago    Up 29 hours   0.0.0.0:33103->3306/tcp   sourcetoad_mariadb103
5bae06984f18   mariadb:10.4                    "docker-entrypoint.s…"   2 weeks ago    Up 29 hours   0.0.0.0:33104->3306/tcp   sourcetoad_mariadb104
6eabb950ac3f   mariadb:10.2                    "docker-entrypoint.s…"   2 weeks ago    Up 29 hours   0.0.0.0:33102->3306/tcp   sourcetoad_mariadb102
81c4e38c7e23   jwilder/nginx-proxy             "/app/docker-entrypo…"   2 weeks ago    Up 29 hours   0.0.0.0:80->80/tcp        sourcetoad_nginx_proxy
234b917d34d5   mariadb:10.5                    "docker-entrypoint.s…"   2 weeks ago    Up 29 hours   0.0.0.0:33105->3306/tcp   sourcetoad_mariadb105
```

We might need a bit of bash, but we know if you give us 33102 we know its maria102, etc. I think if we get to ask about to 2 less things and do some guesses behind the scene. This cuts down the parameters needed to use this.

File: scripts/db_migrate.sh [hunk 2]
```diff
@@ -8,14 +8,15 @@ DB_PORT=''
 DB_HOST='127.0.0.1'
 DB_USER='root'
 DB_PASS='root'
-DB_EXPORT_FLAGS='--routines --column-statistics=0 --quick --hex-blob --single-transaction'
-DB_IMPORT_FLAGS=''
+DB_EXPORT_FLAGS='--routines --quick --hex-blob --single-transaction'
```
[the remark this answers is not in the document]
Comment (chavezellis) [reply scripts/db_migrate.sh:11]: It was causing it to fail. 

`Copying...! 
mysqldump: unknown variable 'column-statistics=0'
Copied...!`

File: scripts/db_migrate.sh [hunk 3]
```diff
@@ -19,6 +19,13 @@ read -rp "Enter (FROM) database name: " DB_NAME
 read -rp "Enter (TO) port number:" DB_TO_PORT
 read -rp "Enter (TO) database name: " DB_TO_NAME
 
+# Given the output of "docker ps" and a port, can find a matching line of:
+# 6eabb950ac3f   mariadb:10.2  "docker-entrypoint.s…" 2 weeks ago Up 30 hours 0.0.0.0:33102->3306/tcp sourcetoad_mariadb102
+# and extract "sourcetoad_mariadb102"
+CONTAINER_NAME_FROM=$(docker ps | grep $DB_PORT | rev | cut -d" " -f1 | rev)
+CONTAINER_NAME_TO=$(docker ps | grep $DB_TO_PORT | rev | cut -d" " -f1 | rev)
+
+
```
Comment (iBotPeaches) [inline scripts/db_migrate.sh:28]: just a double newline here, then im happy.

File: scripts/db_migrate.sh [hunk 4]
```diff
@@ -35,7 +42,7 @@ case $DB_MODE in
         fi
 
         echo -n "Copying...! $NEWLINE"
-        mysqldump "$DB_EXPORT_FLAGS" --user="${DB_USER}" --password="${DB_PASS}" --host="${DB_HOST}" --port="${DB_PORT}" "${DB_NAME}" | mysql "$DB_IMPORT_FLAGS" --user="${DB_USER}" --password="${DB_PASS}" --host="${DB_HOST}" --port="${DB_TO_PORT}" "${DB_TO_NAME}"
+        docker exec -i "$CONTAINER_NAME_FROM" mysqldump $DB_EXPORT_FLAGS --user="${DB_USER}" --password="${DB_PASS}" --port="${DB_PORT}" --databases "${DB_NAME}" | docker exec -i "$CONTAINER_NAME_TO" mysql --user="${DB_USER}" --password="${DB_PASS}" --port="${DB_TO_PORT}" "${DB_TO_NAME}"
```
Comment (iBotPeaches) [inline scripts/db_migrate.sh:45]: I guess we either remove `DB_IMPORT_FLAGS` or add it back to command since its presently unused.

Comment (chavezellis) [reply scripts/db_migrate.sh:44]: Got it removed.

File: scripts/db_migrate.sh (hunk 3 above)
Comment (chavezellis) [reply scripts/db_migrate.sh]: removed

**Review by iBotPeaches — DISMISSED** [review]

**Review by nicomaurer — APPROVED** [review]

**Review by NilsDelaguardia — DISMISSED** [review]

**Review by joeladam518 — DISMISSED** [review]

**Review by Bankamah — DISMISSED** [review]

File: scripts/db_migrate.sh [hunk 5]
```diff
@@ -19,6 +18,12 @@ read -rp "Enter (FROM) database name: " DB_NAME
 read -rp "Enter (TO) port number:" DB_TO_PORT
 read -rp "Enter (TO) database name: " DB_TO_NAME
 
+# Given the output of "docker ps" and a port, can find a matching line of:
+# 6eabb950ac3f   mariadb:10.2  "docker-entrypoint.s…" 2 weeks ago Up 30 hours 0.0.0.0:33102->3306/tcp sourcetoad_mariadb102
+# and extract "sourcetoad_mariadb102"
+CONTAINER_NAME_FROM=$(docker ps | grep "$DB_PORT" | rev | cut -d" " -f1 | rev)
+CONTAINER_NAME_TO=$(docker ps | grep "$DB_TO_PORT" | rev | cut -d" " -f1 | rev)
```
Comment (chase-patterson) [inline scripts/db_migrate.sh:25]: You can make this a little more readable by having docker give you only the columns you want
```shell
docker ps --format '{{.Names}} {{.Ports}}' | grep "$DB_PORT" | cut -d ' ' -f1
```

Comment (chavezellis) [reply scripts/db_migrate.sh:25]: nice that makes it feel much less dirty thanks!

**Review by iBotPeaches — APPROVED** [review]

**Review by Bankamah — APPROVED** [review]

**Review by chase-patterson — APPROVED** [review]

**Review by joeladam518 — APPROVED** [review]

**Review by frankspress — APPROVED** [review]

**Pull request merged**